File: tcbn/python/update/tcbn_update.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
#
#	tcbn_update.py
#
#					Jun/27/2012
#
import	sys
import	json
import	string
import	pytc
#
sys.path.append ('/var/www/data_base/common/python_common')
from dbm_manipulate import dbm_update_proc
# ------------------------------------------------------------
print ("*** 開始 ***")
#
file_tcbn = sys.argv[1]
id_in = sys.argv[2]
population_in = string.atoi (sys.argv[3])
print ("%s\t%d" % (id_in, population_in))
#
db = pytc.BDB(file_tcbn, pytc.BDBOWRITER | pytc.BDBOCREAT)
#
dbm_update_proc	(db,id_in,population_in)
db.close ()
#
print ("*** 終了 ***")
#
# ------------------------------------------------------------

File: tcbn/coffee/update/tcbn_update.js
#! /usr/local/bin/node
// ---------------------------------------------------------------
//	tcbn_update.js
//
//					Jun/16/2011
//
// ---------------------------------------------------------------
var fs = require("fs");
var sys = require('sys');
var TC = require('/var/www/data_base/common/node_common/lib/tokyocabinet');

var text_manipulate=require ("/var/www/data_base/common/node_common/text_manipulate");
// ---------------------------------------------------------------
console.log ("*** 開始 ***");
var id_in=process.argv[2];
var population_in=process.argv[3];

console.log (id_in + "\t" + population_in);

//
// ---------------------------------------------------------------
var BDB = TC.BDB;
var bdb = new BDB;

var file_in = "/var/tmp/tokyo_cabinet/cities.tcb";

var success = bdb.open (file_in, BDB.OWRITER | BDB.OCREAT);
if (!success) throw bdb.errmsg();

	var vv = bdb.get(id_in);
	if (vv === null) throw bdb.errmsg();
	var unit_aa = JSON.parse (vv);
	var out_str = id_in + "\t";
	out_str += unit_aa['name'] + "\t";
	out_str += unit_aa['population'] + "\t";
	out_str += unit_aa['date_mod'];
	sys.puts (out_str);

	unit_aa['population'] = population_in;
	unit_aa['date_mod'] = text_manipulate.get_current_date_proc ();
	var json_out = JSON.stringify (unit_aa);

	console.log (json_out);
	var success = bdb.put (id_in,json_out);
	if (!success) throw bdb.errmsg();

// ---------------------------------------------------------------
console.log ("*** 終了 ***");
// ---------------------------------------------------------------

